pyo3-benches/benches/bench_pyobject.rs

```diff
@@ -12,8 +12,21 @@ fn drop_many_objects(b: &mut Bencher<'_>) {
     });
 }
 
+fn deferred_drop_many_objects(b: &mut Bencher<'_>) {
+    b.iter_batched_ref(
+        || Python::with_gil(|py| (0..1000).map(|_| py.None()).collect::<Vec<_>>()),
+        |objects| {
+            objects.clear();
+            // Trigger deferred drops
+            Python::with_gil(|_| {})
+        },
+        criterion::BatchSize::PerIteration,
+    );
+}
+
 fn criterion_benchmark(c: &mut Criterion) {
     c.bench_function("drop_many_objects", drop_many_objects);
+    c.bench_function("deferred_drop_many_objects", deferred_drop_many_objects);
```

Comment:
This is not an argument against adding this benchmark, but I fear we will need a multi-threaded benchmark to really measure the coherence traffic effects. This will be painful to do as a microbenchmark, but I fear we will be missing the larger picture without it.

Reply:
Agreed, we will probably need a couple of multi threaded benchmarks to model different usage patterns. In the end I had less time today than I was hoping, but I might be able to do some tomorrow.